lib/tasks/set_duplicate_references.rake

```diff
@@ -0,0 +1,27 @@
+desc "Set duplicate references for sales and lettings logs"
+task set_duplicate_references: :environment do
+  SalesLog.filter_by_year(2023).duplicate_sets.each do |duplicate_set|
+    duplicate_set_id = generate_new_duplicate_set_id
+    next if duplicate_set.any? { |log_id| DuplicateLogReference.exists?(log_id:, log_type: "SalesLog") }
+
+    duplicate_set.each do |log_id|
+      DuplicateLogReference.create(log_id:, log_type: "SalesLog", duplicate_set_id:)
+    end
+  end
+
+  LettingsLog.filter_by_year(2023).duplicate_sets.each do |duplicate_set|
+    duplicate_set_id = generate_new_duplicate_set_id
+    next if duplicate_set.any? { |log_id| DuplicateLogReference.exists?(log_id:, log_type: "LettingsLog") }
+
+    duplicate_set.each do |log_id|
+      DuplicateLogReference.create(log_id:, log_type: "LettingsLog", duplicate_set_id:)
+    end
+  end
+end
+
+def generate_new_duplicate_set_id
+  loop do
+    duplicate_set_id = SecureRandom.random_number(1_000_000)
```

Comment:
Hmm, I see what you mean. I think I'm kinda worried about the race condition with incremental values especially when we create there records for the existing logs. And I can't put a uniqueness condition on the db layer cause there would be several records with this ID. Any thoughts?

Reply:
can we set the duplicate set id just to be the lowest id of all the logs in the set?

Reply:
that assumes logs can't be in multiple separate sets which they _shouldn't_ be able to be right?

Reply:
Yeah, but it kinda means that we'd need to recreate the IDs on all of the logs if a log with a lower ID joins the duplicates set. I feel like it might get kinda fiddly

Reply:
I don't think we'd need to recreate the ids (doesn't matter if it's not actually the lowest), I'm just thinking it would be an easy way to get a unique id for each set

Reply:
Or we could make a duplicate set table and just use ids in that table? To be clear I think this is probably fine as is but just seems like it's reinventing unique indexing that we can use db indexes to handle for us

Reply:
I see what you mean with not having to changes the IDs. But what would be the benefit of using log IDs over a random number? Cause for me it feels that if we start using log IDs as an identifier that ties duplicate logs together for anyone who reads this code in the future it might seem like there was a meaning behind it, whereas really we just want a unique number and it shouldn't matter what it is.

Maybe I'm missing something obvious, but given these wouldn't be unique I'm not sure if we can use indexes for that

And in terms of making a duplicate set table, do you mean in addition to the one added in this PR  or just modelling this one differently?